plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/database/DatabaseNavigatorLabelProvider.java

```diff
@@ -102,6 +106,15 @@ public String getText(Object obj) {
             text = labelProvider.getText(obj);
         } else if (obj instanceof DBNNode dbnNode) {
             text = dbnNode.getNodeDisplayName();
+            if (dbnNode.hasChildren(false) && isRelevantNodeType(dbnNode)) {
+                int childCount = 0;
+                try {
+                    childCount = dbnNode.getChildren(new VoidProgressMonitor()).length;
+                } catch (DBException e) {
+                    log.error("Error fetching children for node: " + dbnNode.getNodeDisplayName());
+                }
```

Comment:
This will cause the UI to freeze, as it will load all children upon expanding a node. I suggest showing a counter only for expanded nodes.